tests/bulkdata/bulk_data_delete_tests.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import User
+from arches.app.models.resource import Resource
+from arches.app.models.tile import Tile
+from arches.app.models import models
+from arches.app.utils.data_management.resource_graphs.importer import import_graph as ResourceGraphImporter
+from arches.app.utils.betterJSONSerializer import JSONSerializer, JSONDeserializer
+from yourapp.models import BulkDataDeletion  # Adjust import as necessary
+from tests.arches_test_case import ArchesTestCase
+import os
+import uuid
+# these tests can be run from the command line via
+# python manage.py test tests.bulkdata.bulk_data_delete_tests --settings="tests.test_settings"
+
+class BulkDataDeletionTests(ArchesTestCase):
+    def setUpClass(self):
+        super().setUpClass()
+        # Setup test data here. You might want to create users, resources, tiles, etc.
+        self.user = User.objects.create(username='testuser', password='securepassword')
+        self.bulk_deleter = BulkDataDeletion()
+
+        with open(os.path.join("tests/fixtures/resource_graphs/Search Test Model.json"), "r") as f:
+            archesfile = JSONDeserializer().deserialize(f)
+        ResourceGraphImporter(archesfile["graph"])
+        self.search_model_graphid = "d291a445-fa5f-11e6-afa8-14109fd34195"
+        self.search_model_name_nodeid = "2fe14de3-fa61-11e6-897b-14109fd34195"
+        # Further setup like creating resources and tiles that the test methods will interact with.
+
+    def test_get_number_of_deletions_with_no_resourceids(self):
+        # Test to ensure deletion count is calculated correctly.
+        loadid = str(uuid.uuid4())
+        resourceids = create_test_resources_and_tiles(self.search_model_graphid, self.search_model_name_nodeid, loadid)
+        number_of_resource, number_of_tiles = self.bulk_deleter.get_number_of_deletions(self.search_model_graphid, self.search_model_name_nodeid, None)
+        
+        self.assertIsInstance(number_of_resource, list)
+        self.assertEqual(number_of_tiles, len(resourceids))
+
+    def test_delete_resources(self):
+        # Test to ensure resources are deleted properly.
+        loadid = str(uuid.uuid4())
+        resourceids = create_test_resources_and_tiles(self.search_model_graphid, self.search_model_name_nodeid, loadid)
+        result = self.bulk_deleter.delete_resources(self.user.id, loadid, self.search_model_graphid, resourceids, verbose=True)
```

Comment:
Consider testing with verbose=False so that you don't have to capture the output with `with captured_stdout()` to prevent test clutter. (Testing `pyprind` itself falls under "don't test the framework" IMO).

Reply:
fix in latest